python/sparkdl/graph/builder.py

```diff
@@ -67,17 +69,101 @@ def __exit__(self, *args):
     def run(self, *args, **kargs):
         """
         This method delegate the TensorFlow graph execution
-        to the underlying tf.Session object to perform 
+        to the underlying tf.Session object to perform
         one step of graph computation.
 
         All the parameters are defined according to `tf.Session.run`
-        Reference: https://www.tensorflow.org/api_docs/python/tf/Session#run        
+        Reference: https://www.tensorflow.org/api_docs/python/tf/Session#run
         """
         return self.sess.run(*args, **kargs)
 
+    def asUDF(self, udf_name, fetches, feeds_to_fields_map=None, blocked=False, register=True):
```

Comment:
I had missed that before: this function does not use a full session, just the current graph of computation. Instead of making it a method to the isolated session, make it a top-level function with the first argument being a graph. This function is meant to be low level, and I want to move it to TensorFrames eventually.